Tribler/community/market/core/matching_engine.py

```diff
@@ -0,0 +1,286 @@
+import logging
+from time import time
+
+from Tribler.community.market.core.order import Order
+from Tribler.community.market.core.orderbook import OrderBook
+from Tribler.community.market.core.price import Price
+from Tribler.community.market.core.pricelevel import PriceLevel
+from Tribler.community.market.core.quantity import Quantity
+from Tribler.community.market.core.tickentry import TickEntry
+from Tribler.community.market.core.timestamp import Timestamp
+from Tribler.community.market.core.trade import Trade
+
+
+class MatchingStrategy(object):
```

Comment:
Why not turn this into a formal interface (https://docs.python.org/2/library/abc.html)? I assume this class will be used by other programmers, who will benefit from having a stricter base class 👍